analyzer/tests/unit/test_option_parser.py

```diff
@@ -265,6 +265,32 @@ def test_ignore_flags_gcc(self):
         res = log_parser.parse_options(action)
         self.assertEqual(res.analyzer_options, ["-fsyntax-only"])
 
+    def test_ignore_warning_flags_gcc(self):
+        """-w -Werror flag should be omitted if the compiler gcc."""
+
+        warning_action_gcc = {
+             "directory": "/tmp",
+             "command":
+             "g++ -std=gnu++14 -w -Werror  /tmp/a.cpp -c /tmp/a.cpp",
+             "file": "/tmp/a.cpp"}
+
+        res = log_parser.parse_options(warning_action_gcc)
+
+        self.assertEqual(["-std=gnu++14"], res.analyzer_options)
+
+    def test_ignore_warning_flags_clang(self):
+        """-w -Werror flag should be omitted if the compiler is clang."""
+
+        warning_action_clang = {
+            "directory": "/tmp",
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Probable insecure usage of temp file/directory.](https://app.codacy.com/app/CodeChecker/codechecker/pullRequest?prid=4642761)